From fc991719baeb3ad248c64f1e2c8643e598a1901c Mon Sep 17 00:00:00 2001
From: "Shoutao.Li" <Shoutao.Li@windriver.com>
Date: Fri, 11 Sep 2015 13:44:12 +0800
Subject: [PATCH] Do not load iptables dynamic libraries iptext*

The latest iptables changed the "iptext iptext4 iptext6" from static
library to dynamic shared library.And in firewall3, these libraries
is loaded.

But we don't need to update iptables and the extension libraries could
be useless to firewall3.So delete the using in makefile and sources.

Signed-off-by: Shoutao.Li <Shoutao.Li@windriver.com>
---
 CMakeLists.txt | 6 +++---
 iptables.c     | 8 +++++---
 iptables.h     | 2 ++
 3 files changed, 10 insertions(+), 6 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 22e890d..93cd217 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -7,11 +7,11 @@ SET(iptc_libs   ip4tc)
 
 SET(CMAKE_SHARED_LIBRARY_LINK_C_FLAGS "")
 
-LIST(APPEND ext_libs iptext)
-LIST(APPEND ext_libs iptext4)
+#LIST(APPEND ext_libs iptext)
+#LIST(APPEND ext_libs iptext4)
 
 IF (NOT DISABLE_IPV6)
-  LIST(APPEND ext_libs iptext6)
+  #LIST(APPEND ext_libs iptext6)
   LIST(APPEND iptc_libs ip6tc)
 ELSE()
   ADD_DEFINITIONS(-DDISABLE_IPV6)
diff --git a/iptables.c b/iptables.c
index 0f41fb0..0396ea0 100644
--- a/iptables.c
+++ b/iptables.c
@@ -53,6 +53,7 @@ get_kernel_version(void)
 	kernel_version = 0x10000 * x + 0x100 * y + z;
 }
 
+#if 0
 static void fw3_init_extensions(void)
 {
 	init_extensions();
@@ -62,6 +63,7 @@ static void fw3_init_extensions(void)
 	init_extensions6();
 #endif
 }
+#endif
 
 struct fw3_ipt_handle *
 fw3_ipt_open(enum fw3_family family, enum fw3_table table)
@@ -100,7 +102,7 @@ fw3_ipt_open(enum fw3_family family, enum fw3_table table)
 	}
 
 	fw3_xt_reset();
-	fw3_init_extensions();
+	//fw3_init_extensions();
 
 	return h;
 }
@@ -374,10 +376,10 @@ load_extension(struct fw3_ipt_handle *h, const char *name)
 	void *lib, **tmp;
 	const char *pfx = (h->family == FW3_FAMILY_V6) ? "libip6t" : "libipt";
 
-	snprintf(path, sizeof(path), "/usr/lib/iptables/libxt_%s.so", name);
+	snprintf(path, sizeof(path), "/usr/lib/xtables/libxt_%s.so", name);
 	if (!(lib = dlopen(path, RTLD_NOW)))
 	{
-		snprintf(path, sizeof(path), "/usr/lib/iptables/%s_%s.so", pfx, name);
+		snprintf(path, sizeof(path), "/usr/lib/xtables/%s_%s.so", pfx, name);
 		lib = dlopen(path, RTLD_NOW);
 	}
 
diff --git a/iptables.h b/iptables.h
index fabefa8..66add19 100644
--- a/iptables.h
+++ b/iptables.h
@@ -39,10 +39,12 @@
 # error "Unsupported xtables version"
 #endif
 
+#if 0
 /* libipt*ext.so interfaces */
 extern void init_extensions(void);
 extern void init_extensions4(void);
 extern void init_extensions6(void);
+#endif
 
 /* Required by certain extensions like SNAT and DNAT */
 extern int kernel_version;
-- 
1.9.1

